rpc/src/v1/helpers/signature.rs

```diff
@@ -0,0 +1,31 @@
+use ethkey::{recover, public_to_address, Signature};
+use jsonrpc_core::Result;
+use v1::types::{Bytes, H520, RichBasicAccount, BasicAccount};
+use v1::helpers::errors;
+use v1::helpers::dispatch::eth_data_hash;
+use hash::keccak;
+
+pub fn verify_signature(is_prefixed: bool, message: Bytes, signature: H520, chain_id: Option<u64>) -> Result<RichBasicAccount> {
+	let hash = if is_prefixed {
+		eth_data_hash(message.0)
+	} else {
+		keccak(message.0.clone())
+	};
+
+	let signature = Signature::from(signature.0);
+	let is_valid_for_current_chain = chain_id.map(|chain_id| {
+		let v = signature.v() as u64;
+		if v > 36  {
+			let decoded_v = (v - 35) - (chain_id * 2);
```

Comment:
this can still overflow ```if chain_id*2 is bigger than v-35``` (maybe only theoretical but still ...)

Consider using checked_sub and saturated_mul for this use-case instead!

Reply:
not sure why we need saturating_mul with chain_id, isn't it supposed to be a compile-time constant?